File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.ts
function makeFancyString(s: string): string {
    let newS = '', counter = 0;
    for(let i = 0; i < s.length; i++) {
        if(s[i] === s[i - 1]) counter++
        else counter = 1;
        if(counter <= 2) newS += s[i]
    }
    return newS
};

File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.rs
impl Solution {
    pub fn make_fancy_string(s: String) -> String {
        let mut res = String::new();
        let mut prev_char: Option<char> = None;
        let mut count = 0;
        for c in s.chars() {
            if Some(c) == prev_char {
                count += 1;
            } else {
                count = 1;
            }
            if count < 3 {
                res.push(c);
            }
            prev_char = Some(c);
        }
        res
    }
}